packages/edit-site/src/components/post-router/update-query-params.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Updates the query args in the window to match the given context.
+ *
+ * @param {string} contextType Current context of the site editor,
+ * 'wp_template', 'wp_template_part', or 'content'.
+ * @param {number|string} identifier Identifier for the given context.
+ * This may be a postId for 'wp_template' or 'wp_template_part', or may
+ * be a JSON string representing the 'page' state in the edit-site store
+ * for various content contexts.
+ */
+export default function updateQueryParams( contextType, identifier ) {
+	const queryParams = new URLSearchParams( window.location.search );
```

Comment:
[`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) – it looks like it isn't supported in IE at all, so it should be polyfilled.

There is also [`@wordpress/url`](https://github.com/WordPress/gutenberg/tree/master/packages/url) package that is used in other places and it has some WordPress specific methods. I have no idea if it's applicable here, but I though I would share.

Reply:
Good point. `getQueryArg` and `addQueryArgs` should be enough for this use-case.

Reply:
Good idea, switched to use @wordpress/url for all the use cases in this PR